test/e2e_node/util.go

```diff
@@ -623,3 +624,22 @@ func WaitForPodInitContainerToFail(ctx context.Context, c clientset.Interface, n
 		return false, nil
 	})
 }
+
+func nodeNameOrIP() string {
```

Comment:
there are other instances:

- `logKubeletLatencyMetrics` uses `fmt.Sprintf("%s:%d", framework.TestContext.NodeName, ports.KubeletReadOnlyPort), "/metrics")`
- `getPodFromStandaloneKubelet` uses `endpoint := fmt.Sprintf("http://127.0.0.1:%d/pods", ports.KubeletReadOnlyPort)`

Would it be more reliable to use localhost here as well? If you can align those two usages as well, would be great

Reply:
I am comfortable fixing `logKubeletLatencyMetrics` but want to leave the `getPodFromStandaloneKubelet` alone, until i see evidence that we need to fix that. ok?

Reply:
Done!

Reply:
Would switching to `localhost` everywhere be better than adding this name resolution logic?

Reply:
@SergeyKanzhelev one reason i did this here was to ensure that i am not adversely affecting existing tests that may have an assumption on the node name or its ip (and falling back on the interface ONLY if resolution fails). I am happy to revisit this again to switch everything to `localhost`

Reply:
/lgtm

Reply:
I recommend localhost, if possible , too